File: 25.cpp

struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(nullptr) {
	}
};
class Solution {
public:
    ListNode* Merge(ListNode* pHead1, ListNode* pHead2)
    {
        if(pHead1== nullptr){
            return pHead2;
        }
        if(pHead2== nullptr){
            return pHead1;
        }
        ListNode* mergerPhead = nullptr;
        if(pHead1->val<pHead2->val){
            mergerPhead = pHead1;
            mergerPhead->next = Merge(pHead1->next,pHead2);
        } else{
            mergerPhead = pHead2;
            mergerPhead->next = Merge(pHead1,pHead2->next);
        }
        return mergerPhead;

    }
};

File: 25.py
# -*- coding:utf-8 -*-
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None
class Solution:
    # 返回合并后列表
    def init(self,pHead1:ListNode, pHead2:ListNode):
        self.pHead1:ListNode = pHead1
        self.pHead2:ListNode = pHead2
        if(self.pHead1.val<self.pHead2.val):
            self.pHead = self.pHead1
            self.pHead1 = self.pHead1.next
        else:
            self.pHead = self.pHead2
            self.pHead2 = self.pHead2.next
        self.pNode = self.pHead

    def core(self):
        if(self.pHead1.val<self.pHead2.val):
            self.pNode.next = self.pHead1
            self.pHead1 = self.pHead1.next
            self.pNode = self.pNode.next
        else:
            self.pNode.next = self.pHead2
            self.pNode = self.pNode.next
            self.pHead2 = self.pHead2.next

    def Merge(self, pHead1:ListNode, pHead2:ListNode):
        if(pHead1==None and pHead2==None):
            return None
        self.init(pHead1,pHead2)
        while(self.pHead1!=None or self.pHead2!=None):
            self.core()
        if(self.pHead1==None):
            self.pNode.next = self.pHead2
        else:
            self.pNode.next = self.pHead1
        return self.pHead

# write code here